frontend/src/components/Search/SearchContext.tsx

```diff
@@ -107,6 +116,27 @@ export const SearchContext: React.FC<Props> = ({
 	const [aiQuery, setAiQuery] = useState('')
 	const { queryParts, tokens } = parseSearch(query)
 	const tokenGroups = buildTokenGroups(tokens)
+	const {
+		handleSearch,
+		recentSearches,
+		activeTab,
+		historyLoading,
+		setActiveTab,
+	} = useSearchHistory()
+	const handleSubmit = (query: string) => {
+		onSubmit(query)
+		if (query) {
+			const queryParts = parseSearch(query)?.queryParts || []
+			try {
+				const newQueryParts = JSON.parse(
+					JSON.stringify(queryParts || []),
+				)
+				handleSearch?.(query, newQueryParts)
+			} catch (err) {
+				//do nothing
```

Comment:
a `console.error` might be nice here in case something breaks in the future with the history format